rest/data/host.go

```diff
@@ -404,3 +417,26 @@ func transitionIntentHostToDecommissioned(ctx context.Context, env evergreen.Env
 
 	return nil
 }
+
+// setReadyForReprovisioning marks the host as ready to reprovision. This does
+// not modify quarantined hosts, because quarantined hosts cannot be
+// reprovisioned (and are therefore not ready to reprovision). If a quarantined
+// host needs to reprovision, it will automatically do so when it's brought out
+// of quarantine.
+func setReadyForReprovisioning(ctx context.Context, env evergreen.Environment, h *host.Host) error {
+	if h.NeedsReprovision == host.ReprovisionNone {
+		return nil
+	}
+
+	if utility.StringSliceContains([]string{evergreen.HostProvisioning, evergreen.HostRunning}, h.Status) {
```

Comment:
Should we also consider `HostStopping`?

Reply:
Only spawn hosts are really stoppable. For task hosts, they can be quarantined or terminated, but we don't have a reason to ever stop them, so nothing in the task host lifecycle handles that.